rust/src/binary_view_event_handler.rs

```diff
@@ -0,0 +1,66 @@
+use crate::binaryview::BinaryView;
+use binaryninjacore_sys::{
+    BNBinaryView, BNBinaryViewEventType, BNNewViewReference, BNRegisterBinaryViewEvent,
+};
+
+pub trait BinaryViewEventHandler: 'static + Sync {
+    fn on_event(&mut self, binary_view: &BinaryView);
```

Comment:
I don't think this should be mut? If you need mut, there should probably be a non-mut version? But I think everything should be able to be done in a non-mut way....

Reply:
Hmm, upon further consideration I agree this instance should be &self to agree with `Command` and friends. The `&mut` makes more sense if we know for certain that binary ninja isn't going to call the fn at the same time from multiple threads and given `Sync we are not making that assumption.